### PR TITLE
Project link commands

### PR DESCRIPTION
**Pull request by james-wallis**
Should be merged after https://github.com/eclipse/codewind/pull/2667
## What type of PR is this ? 

- [ ] Bug fix
- [x] Enhancement

## What does this PR do ?
Adds `cwctl project link` commands to call the PFE API
    * Will only work with https://github.com/eclipse/codewind/pull/2667 and not the first drop.
Adds tests for them

### CODE REVIEW

File: pkg/actions/commands.go [hunk 1]
```diff
@@ -217,6 +217,63 @@ func Commands() {
...
+				{
+					Name:  "link",
+					Usage: "Manage project links",
+					Subcommands: []cli.Command{
+						{
+							Name:    "list",
+							Aliases: []string{"ls"},
+							Usage:   "Lists all the links for a project",
+							Flags: []cli.Flag{
+								cli.StringFlag{Name: "id,i", Usage: "Project ID", Required: true},
```
Comment (jcockbain) [inline pkg/actions/commands.go:229]: big nit: spaces after commas in the flags (like the other commands)

File: pkg/actions/project.go [hunk 2]
```diff
@@ -284,3 +284,155 @@ func ProjectRestart(c *cli.Context) {
...
+		os.Exit(1)
+	}
+
+	projectLinkErr := project.CreateProjectLink(http.DefaultClient, conInfo, conURL, projectID, targetProjectID, envName)
+	if projectLinkErr != nil {
+		HandleProjectError(projectLinkErr)
+		os.Exit(1)
+	}
+
+	response, _ := json.Marshal(project.Result{Status: "OK", StatusMessage: "Project link create request accepted"})
```
Comment (jcockbain) [inline pkg/actions/project.go:366]: do you need to consider `--json` here (if it's not set)?

File: pkg/project/link.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	}
+	// LinkParameters : The request structure to create a link
+	LinkParameters struct {
+		EnvName         string `json:"envName"`
+		TargetProjectID string `json:"targetProjectID,omitempty"`
+		UpdatedEnvName  string `json:"updatedEnvName,omitempty"`
+	}
+)
+
+// GetProjectLinks calls the project link API on PFE with a POST request
```
Comment (jcockbain) [inline pkg/project/link.go:40]: links API?

File: pkg/project/link_test.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+var errInvalidRequest = errors.New(textInvalidRequest)
+var errTargetNotFound = errors.New(textProjectLinkTargetNotFound)
+var errConflict = errors.New(textProjectLinkConflict)
+var errUnknownHTTPCode = errors.New(textUnknownResponseCode)
+
+func TestGetProjectLinks(t *testing.T) {
+	t.Run("Expect success - project links should be returned", func(t *testing.T) {
```
Comment (jcockbain) [inline pkg/project/link_test.go:33]: test for the error status codes in get?

File: pkg/project/link_test.go [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+		mockConnection := connections.Connection{ID: "local"}
+
+		projectLinkErr := UpdateProjectLink(mockClient, &mockConnection, "dummyurl", "dummyProjectID", "dummyEnvName", "dummyUpdatedEnvName")
+		wantedError := &ProjectError{errOpResponse, errUnknownHTTPCode, errUnknownHTTPCode.Error()}
+		assert.Equal(t, wantedError, projectLinkErr)
+	})
+}
+
+func TestDeleteProjectLinks(t *testing.T) {
+	t.Run("Expect success - project links should be created", func(t *testing.T) {
```
Comment (jcockbain) [inline pkg/project/link_test.go:138]: copy and paste is not your friend (same as above)

File: pkg/project/link_test.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+		mockConnection := connections.Connection{ID: "local"}
+
+		projectLinkErr := CreateProjectLink(mockClient, &mockConnection, "dummyurl", "dummyProjectID", "dummyTargetProjectID", "dummyEnvName")
+		wantedError := &ProjectError{errOpResponse, errUnknownHTTPCode, errUnknownHTTPCode.Error()}
+		assert.Equal(t, wantedError, projectLinkErr)
+	})
+}
+
+func TestUpdateProjectLinks(t *testing.T) {
+	t.Run("Expect success - project links should be created", func(t *testing.T) {
```
Comment (jcockbain) [inline pkg/project/link_test.go:96]: updated?

File: pkg/actions/project.go [hunk 7]
```diff
@@ -284,3 +284,155 @@ func ProjectRestart(c *cli.Context) {
 	fmt.Println(string(response))
 	os.Exit(0)
 }
+
+// ProjectLinkList : lists all the links for a project
+func ProjectLinkList(c *cli.Context) {
+	projectID := strings.TrimSpace(strings.ToLower(c.String("id")))
+
+	conID, getConnectionIDErr := project.GetConnectionID(projectID)
```
Comment (jcockbain) [inline pkg/actions/project.go:292]: not a you thing, but noticed testing the commands that when I give a non existent project I get a file read error (in /.codewind/connections). We could perhaps catch this and return something clearer.

File: pkg/actions/commands.go (hunk 1 above)
Comment (james-wallis) [reply pkg/actions/commands.go:229]: I've done this but fyi some others don't have spaces

File: pkg/actions/project.go (hunk 2 above)
Comment (james-wallis) [reply pkg/actions/project.go:366]: I've followed the general layout of the commands in `project.go` and most just print the response

**Pull request merged**